### PR TITLE
Add recipe for moz-sql-parser

### PR DESCRIPTION
**Pull request by xhochy**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for review/merge.

File: recipes/moz-sql-parser/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv "
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - mo-future >=2.48.19205
+    - pyparsing ==2.3.1
```
Comment (beckermr) [inline recipes/moz-sql-parser/meta.yaml]: This pin is rather exact. Is it needed?

Comment (xhochy) [reply recipes/moz-sql-parser/meta.yaml]: Not sure, this is the pin upstream uses and I would just mirror their requirements here.

Comment (beckermr) [reply recipes/moz-sql-parser/meta.yaml]: Wow ok. LGTM then.

File: recipes/moz-sql-parser/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - python
+
+test:
+  imports:
+    - moz_sql_parser
+
+about:
+  home: "https://github.com/mozilla/moz-sql-parser"
+  license: "Mozilla Public 2.0 (MPL 2.0)"
```
Comment (chrisburr) [inline recipes/moz-sql-parser/meta.yaml]: It's nice to use the SPDX identifiers to try and make the metadata consistent across conda-forge:
```suggestion
  license: MPL-2.0
```

File: recipes/mo-future/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  run:
+    - python
+
+test:
+  imports:
+    - mo_future
+
+about:
+  home: "https://github.com/klahnakoski/mo-files"
+  license: "Mozilla Public 2.0 (MPL 2.0)"
```
Comment (chrisburr) [inline recipes/mo-future/meta.yaml]:
```suggestion
  license: MPL-2.0
```

**Pull request merged**